mqtt_client.c

```diff
@@ -1168,7 +1168,9 @@ esp_err_t esp_mqtt_client_stop(esp_mqtt_client_handle_t client)
         }
 
         client->run = false;
+		MQTT_API_UNLOCK_FROM_OTHER_TASK(client);
         xEventGroupWaitBits(client->status_bits, STOPPED_BIT, false, true, portMAX_DELAY);
+		MQTT_API_LOCK_FROM_OTHER_TASK(client);
         client->state = MQTT_STATE_UNKNOWN;
```

Comment:
I would suggest simplifying this fix a little by moving this line just after `client->run = false;` and unlocking the mutex only once. This should be faster and maybe also a bit safer too (invalidating client's state while taking the last loop iteration in client task)

Reply:
Thanks for the suggestion, it makes a lot more sense that way!